cicd/_common_deploy_logic.sh

```diff
@@ -74,7 +74,14 @@ function teardown {
     for ns in ${UNIQUE_NAMESPACES[@]}; do
         echo "Running teardown for ns: $ns"
         set +e
-        collect_k8s_artifacts $ns
+
+        if check_junit_files "${ARTIFACTS_DIR}/junit-*.xml"; then
```

Comment:
We should also check if we arrived here due to an ERR signal and not a normal/successful EXIT signal

If we arrived here due to an error, we should also collect logs during teardown.

Reply:
I did the proxy-trap function thing we discussed - I now check if we arrived there by anything different than a EXIT then it'll just collect artifacts. Check the current implementation and LMK what you think

Reply:
The proxy trap is a cool solution